src/module/ovs/event.rs

```diff
@@ -321,15 +323,103 @@ pub(crate) struct ActionEvent {
+            OvsAction::PushMpls => write!(f, " push_mpls")?,
+            OvsAction::PopMpls => write!(f, " pop_mpls")?,
+            OvsAction::SetMasked => write!(f, " set_masked")?,
+            OvsAction::Ct(ct) => {
+                write!(f, " ct zone {}", ct.zone_id)?;
+
+                if let Some(nat) = &ct.nat {
+                    write!(f, " nat")?;
+                    if let Some(dir) = &nat.dir {
+                        match dir {
+                            NatDirection::Src => write!(f, "(src")?,
+                            NatDirection::Dst => write!(f, "(dst")?,
+                        }
+
+                        if ct.flags & R_OVS_CT_NAT_RANGE_MAP_IPS != 0 {
+                            if let (Some(min_addr), Some(max_addr)) =
+                                (nat.min_addr.as_ref(), nat.max_addr.as_ref())
+                            {
+                                if min_addr.eq(max_addr) {
+                                    write!(f, "={}", min_addr)?;
+                                } else {
+                                    write!(f, "={}-{}", min_addr, max_addr)?;
+                                }
+                            }
+                        }
+                        if ct.flags & R_OVS_CT_NAT_RANGE_PROTO_SPECIFIED != 0 {
+                            if let (Some(min_port), Some(max_port)) =
+                                (nat.min_port.as_ref(), nat.max_port.as_ref())
+                            {
+                                if min_port.eq(max_port) {
+                                    write!(f, ":{}", min_port)?;
+                                } else {
+                                    write!(f, ":{}-{}", min_port, max_port)?;
+                                }
+                            }
+                        }
+                        write!(f, ")")?;
+                    }
+                }
+
+                if ct.is_commit()
+                    || ct.is_force()
+                    || ct.is_persistent()
+                    || ct.is_hash()
+                    || ct.is_random()
+                {
+                    write!(f, " ")?;
+                    let mut comma = DelimWriter::new(',');
+                    if ct.is_commit() {
+                        comma.write(f)?;
+                        write!(f, "commit")?;
+                    }
+                    if ct.is_force() {
+                        comma.write(f)?;
+                        write!(f, "force")?;
+                    }
+                    if ct.is_persistent() {
+                        comma.write(f)?;
+                        write!(f, "presistent")?;
+                    }
+                    if ct.is_hash() {
+                        comma.write(f)?;
+                        write!(f, "hash")?;
+                    }
+                    if ct.is_random() {
+                        comma.write(f)?;
+                        write!(f, "random")?;
+                    }
```

Comment:
xD, I used join before but then you suggested:

> In general for this whole function I would advise to use write!(f, ...) directly when constructing the output to avoid using format! whenever possible.